src/pages/leaderboard/ui/table.tsx

```diff
@@ -0,0 +1,175 @@
+'use client';
+
+import { useAutoAnimate } from '@formkit/auto-animate/react';
+import { useState } from 'react';
+import cn from 'clsx';
+import { Text } from '@penumbra-zone/ui/Text';
+import { TableCell } from '@penumbra-zone/ui/TableCell';
+import { SegmentedControl } from '@penumbra-zone/ui/SegmentedControl';
+import { ValueViewComponent } from '@penumbra-zone/ui/ValueView';
+import Link from 'next/link';
+import { SquareArrowOutUpRight } from 'lucide-react';
+import {
+  AssetSelector,
+  AssetSelectorValue,
+  isBalancesResponse,
+} from '@penumbra-zone/ui/AssetSelector';
+import { useAssets } from '@/shared/api/assets';
+import { useBalances } from '@/shared/api/balances';
+import { getMetadataFromBalancesResponse } from '@penumbra-zone/getters/balances-response';
+import { useLeaderboard } from '@/pages/leaderboard/api/use-leaderboard';
+import { DEFAULT_INTERVAL, LeaderboardIntervalFilter } from '../api/utils';
+import { uint8ArrayToHex } from '@penumbra-zone/types/hex';
+import { Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+
+const getAssetId = (value: AssetSelectorValue | undefined): string | undefined => {
+  if (!value) {
+    return undefined;
+  }
+
+  const metadata: Metadata = isBalancesResponse(value)
+    ? getMetadataFromBalancesResponse(value)
+    : value;
+
+  return metadata.penumbraAssetId?.inner
+    ? uint8ArrayToHex(metadata.penumbraAssetId.inner)
+    : undefined;
+};
+
+export const LeaderboardTable = () => {
+  const [parent] = useAutoAnimate();
+
+  const [interval, setInterval] = useState(DEFAULT_INTERVAL);
+  const [base, setBase] = useState<AssetSelectorValue>();
+  const [quote, setQuote] = useState<AssetSelectorValue>();
+
+  const baseAssetId = getAssetId(base);
+  const quoteAssetId = getAssetId(quote);
+
+  const {
+    data: leaderboard,
+    error,
+    isLoading,
+  } = useLeaderboard({
+    limit: 30,
+    interval,
+    base: baseAssetId,
+    quote: quoteAssetId,
+  });
+
+  const { data: assets } = useAssets();
+  const { data: balances } = useBalances();
+
+  if (error) {
+    return (
+      <Text large color='destructive.light'>
+        {error.message}
+      </Text>
+    );
+  }
+
+  const { data: positions } = leaderboard ?? {};
+
+  return (
```

Comment:
todo: this page needs a loading state with skeletons, right now it's empty for long time

Reply:
Added it.